### PR TITLE
Fiat exchange rate service

### PR DESCRIPTION
**Pull request by isocolsky**

### CODE REVIEW

**Comment by isocolsky** [comment]
rebased

File: lib/storage.js [hunk 1]
```diff
@@ -572,6 +573,38 @@ Storage.prototype.fetchActiveAddresses = function(walletId, cb) {
...
+  }, cb);
+};
+
+Storage.prototype.fetchFiatRate = function(providerName, code, ts, cb) {
+  var self = this;
+  self.db.collection(collections.FIAT_RATES).find({
+    provider: providerName,
+    code: code,
+    ts: {
+      $lte: ts
```
Comment (matiu) [inline lib/storage.js]: should be some kind of limit for on ts difference? (like a day of so?)

File: test/integration/fiatrateservice.js [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+          value: value,
+        }, {
+          code: 'EUR',
+          value: value,
+        }], next);
+      }, function(err) {
+        should.not.exist(err);
+        service.getRate({
+          code: 'USD',
+          ts: [50, 100, 199, 500],
```
Comment (matiu) [inline test/integration/fiatrateservice.js]: very nice! :)

**Comment by matiu** [comment]
ACK. Great job!! :)

File: lib/storage.js (hunk 1 above)
Comment (isocolsky) [inline lib/storage.js]: The response sent to the client contains a `fetchedOn` field which should let the client app decide on the validity of the rate. Anyways, I think you are right that we have to limit this server-side also.

**Pull request merged**